qiskit_experiments/calibration/experiments/drag.py

```diff
@@ -0,0 +1,246 @@
+
+        Raises:
+            CalibrationError:
+                - If the beta parameters in the xp and xm pulses are not the same.
+                - If either the xp or xm pulse do not have at least one Drag pulse.
+                - If the number of different repetition series is not three.
+        """
+        # TODO this is temporary logic.
+        self.set_analysis_options(
+            data_processor=get_to_signal_processor(
+                meas_level=self.run_options.meas_level,
+                meas_return=self.run_options.meas_return,
+            ),
+        )
+
+        plus_sched = self.experiment_options.rp
+        minus_sched = self.experiment_options.rm
+
+        if plus_sched is None:
+            beta = Parameter("β")
+            with pulse.build(backend=backend, name="xp") as plus_sched:
+                pulse.play(
+                    pulse.Drag(
+                        duration=self.experiment_options.duration,
+                        amp=self.experiment_options.amp,
+                        sigma=self.experiment_options.sigma,
+                        beta=beta,
+                    ),
+                    pulse.DriveChannel(self._physical_qubits[0]),
+                )
+
+            with pulse.build(backend=backend, name="xm") as minus_sched:
+                pulse.play(
+                    pulse.Drag(
+                        duration=self.experiment_options.duration,
+                        amp=-self.experiment_options.amp,
+                        sigma=self.experiment_options.sigma,
+                        beta=beta,
+                    ),
+                    pulse.DriveChannel(self._physical_qubits[0]),
+                )
+
+        if len(plus_sched.parameters) != 1 or len(minus_sched.parameters) != 1:
+            raise CalibrationError(
+                "The schedules for Drag calibration must both have one free parameter."
+                f"Found {len(plus_sched.parameters)} and {len(minus_sched.parameters)} "
+                "for Rp and Rm, respectively."
+            )
+
+        beta_xp = next(iter(plus_sched.parameters))
+        beta_xm = next(iter(minus_sched.parameters))
+
+        if beta_xp != beta_xm:
+            raise CalibrationError(
+                f"Beta for xp and xm in {self.__class__.__name__} calibration are not identical."
+            )
+
+        xp_gate = Gate(name="Rp", num_qubits=1, params=[beta_xp])
+        xm_gate = Gate(name="Rm", num_qubits=1, params=[beta_xp])
+
+        reps = self.experiment_options.reps
+        if len(reps) != 3:
+            raise CalibrationError(
+                f"{self.__class__.__name__} must use exactly three repetition numbers. "
+                f"Received {reps} with length {len(reps)} != 3."
+            )
+
+        circuits = []
+        for beta in self.experiment_options.betas:
+
+            beta = np.round(beta, decimals=6)
+
+            for idx, rep in enumerate(reps):
+                circuit = QuantumCircuit(1)
```

Comment:
you don't need to regenerate circuit in each loop. parameter is automatically passed to the schedule attached to the circuit.

Reply:
See here: e485440